OpenRA.Game/Scripting/ScriptTypes.cs

```diff
@@ -149,23 +148,25 @@ public static bool TryGetClrValue(this LuaValue value, Type t, out object clrObj
 
 		public static LuaValue ToLuaValue(this object obj, ScriptContext context)
 		{
-			if (obj is LuaValue)
-				return (LuaValue)obj;
+			{
+				if (obj is LuaValue v)
+					return v;
 
-			if (obj == null)
-				return LuaNil.Instance;
+				if (obj == null)
+					return LuaNil.Instance;
 
-			if (obj is double)
-				return (double)obj;
+				if (obj is double d)
+					return d;
 
-			if (obj is int)
-				return (int)obj;
+				if (obj is int i)
+					return i;
 
-			if (obj is bool)
-				return (bool)obj;
+				if (obj is bool b)
+					return b;
 
-			if (obj is string)
-				return (string)obj;
+				if (obj is string s)
+					return s;
+			}
```

Comment:
Doesn't this affect performance negatively?

Reply:
I have not heard anything about pattern matching causing negative perf for casts, do you have a link about anything like that?

Reply:
I meant that we are creating a bunch of variables instead of using just 1. Though it may be already be happening under the hood and not exposed, I dunno that

Reply:
I suspect that, if at all, it saves a cast.

Reply:
Given we're in Lua-land any tiny perf cost or benefit from this is going to be noise compared to invoking functions in another runtime, so I think it's safe to not worry.